### DISCUSSION TITLE
[@types/node] TLSSocket constructor may take a Duplex

### DISCUSSION THREAD

**Discussion by hildjj** [DefinitelyTyped/DefinitelyTyped#56377] [Issues with a @types package] [answered]
According to the nodejs [docs](https://nodejs.org/api/tls.html#tls_new_tls_tlssocket_socket_options), the TLSSocket constructor's first argument may be `net.Socket |  stream.Duplex`.  @node/types 16.10.3 currently says:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/7aded252f5659f98dad25d2a1b3d04f8e324ab8c/types/node/tls.d.ts#L143

I propose that this be changed to:

```js
constructor(socket: net.Socket | stream.Duplex, options?: TLSSocketOptions);
```

My use case: I am using Duplex sockets for mocking a TLS server instance.

[a comment by an automated account is not shown]

**Comment by ghost** [comment, marked as answer]
Fixed in PR #68383
